File: 백준/Silver/4948. 베르트랑 공준/베르트랑 공준.java
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder answer = new StringBuilder();
        while (true) {
            int n = Integer.parseInt(br.readLine());
            if (n == 0)
                break;
            int count = 0;
            for(int i = n + 1; i <= 2 * n; i++){
                boolean primeCheck = true;
                for(int j = 2; j < (int)Math.sqrt((double)i) + 1; j++){
                    if(i % j == 0){
                        primeCheck = false;
                        break;
                    }
                }
                if(primeCheck) count++;
            }
            answer.append(count).append("\n");
        }
        br.close();
        System.out.print(answer.toString());
    }
}

File: 백준/Silver/4948. 베르트랑 공준/베르트랑 공준.py
import sys, math
input = sys.stdin.readline

def eratosChe(n):
    limit = 2 * n
    isPrimeNums = [True] * (limit + 1)
    isPrimeNums[0] = False
    isPrimeNums[1] = False
    for i in range(2, int(math.sqrt(limit)) + 1):
        if isPrimeNums[i]:
            for j in range(i * i, limit + 1, i):
                isPrimeNums[j] = False
    
    count = 0
    for i in range(n+1, 2*n+1):
        if isPrimeNums[i]:
            count += 1

    return count

while True:
    n = int(input())
    if n == 0:
        break
    sys.stdout.write(str(eratosChe(n)) + "\n")